components/ActivityCard/index.js

```diff
@@ -11,22 +12,22 @@ export default function ActivityCard({
 }) {
   return (
     <StyledArticle>
+      <ImageContainer>
+        <StyledImageComponent images={activity.images} alt={activity.title} />
+        <StyledFavoriteButton
+          onClick={(event) => {
+            event.preventDefault();
+            onToggleFavorite(activity.id);
+          }}
```

Comment:
You should not need to prevent the default behaviour here:

```suggestion
          onClick={() => {
            onToggleFavorite(activity.id);
          }}
```